orchagent/bufferorch.cpp

```diff
@@ -715,7 +716,18 @@ task_process_status BufferOrch::processBufferProfile(KeyOpFieldsValuesTuple &tup
                 }
                 else if (SAI_STATUS_SUCCESS != sai_status)
                 {
-                    SWSS_LOG_ERROR("Failed to modify buffer profile, name:%s, sai object:%" PRIx64 ", status:%d", object_name.c_str(), sai_object, sai_status);
+                    SWSS_LOG_NOTICE("Unable to modify buffer profile, name:%s, sai object:%" PRIx64 ", status:%d, will retry one more time", object_name.c_str(), sai_object, sai_status);
+                    attribs_to_retry.push_back(attribute);
+                }
+            }
+
+            for (auto &attribute : attribs)
```

Comment:
This retry logic is needed for specific buffer profile attributes. Instead of just retrying blindly, why can't we handle this combination (xoff, xon and size) the same way as we did for  wred attributes - min and max (https://github.com/sonic-net/sonic-swss/blob/master/orchagent/qosorch.cpp#L595)

Reply:
Hi @neethajohn 
Unlike the wred/ecn profile which has an explicit logic shared by all vendors, the buffer profile logic can differ among various vendors. In this sense, we can’t find a common logic but prefer to retry here and leave the sanity check to the vendor SAI.